src/lib.rs

```diff
@@ -174,8 +176,70 @@ fn enum_variant_type_impl(ast: DeriveInput) -> proc_macro2::TokenStream {
     let data_enum = data_enum(&ast);
     let variants = &data_enum.variants;
 
+    let mut wrap_in_module = None::<Ident>;
+    let mut derive_for_all_variants = None::<Attribute>;
+    let mut marker_trait_ident = None::<Ident>;
+
+    for attr in ast.attrs.iter() {
+        if attr.path.is_ident("evt") {
+            if let Ok(Meta::List(list)) = attr.parse_meta() {
+                for item in list.nested.iter() {
+                    match item {
+                        NestedMeta::Meta(Meta::NameValue(name_value)) => {
+                            if let (true, Lit::Str(lit_str)) =
+                                (name_value.path.is_ident("module"), &name_value.lit)
+                            {
+                                wrap_in_module =
+                                    Some(Ident::new(&lit_str.value(), Span::call_site()));
+                            } else {
+                                panic!("Expected #[evt(module = \"some_module_name\")]");
+                            }
+                        }
+                        NestedMeta::Meta(Meta::List(list)) => {
+                            if list.path.is_ident("derive") {
+                                let items = list
+                                    .nested
+                                    .iter()
+                                    .map(|nested_meta| {
+                                        if let NestedMeta::Meta(Meta::Path(path)) = nested_meta {
+                                            path.clone()
+                                        } else {
+                                            panic!("Expected #[evt(derive(Clone, Debug))]");
+                                        }
+                                    });
+                                derive_for_all_variants = Some(parse_quote! {
+                                    #[derive( #(#items),* )]
+                                });
+                            } else if list.path.is_ident("derive_marker_trait") {
+                                if list.nested.len() != 1 {
```

Comment:
eh, what if, someone wants to derive multiple marker traits?

with this guard they can do this :smile:

```rust
#[evt(derive_marker_trait(TraitA))]
#[evt(derive_marker_trait(TraitB))]
```

might as well allow them the shorthand

Reply:
Haha I was debating this. Was wondering about what use case would require two marker traits given that the traits and implementoreers would be exactly the same? Decided it was more of anti-pattern and not to permit it. Will change to allow for a shorthand version if there is a good reason...?

Reply:
hm, use case I can think of is marker traits from two separate crates

other way round, I can't think of many cases where preventing it protects the user very much -- specifying the same marker trait in an attribute should be rare, and even if they do, the rust compilation error would probably be clear enough (multiple conflicting implementations)

Reply:
Ah there is a misunderstanding. `#[evt(derive_marker_trait(MyTrait))` actually generates a new marker trait `MyTrait` definition (e.g. it generates `trait MyTrait {}`, it is not a reference to a existing trait just the name of a new one to generate.
https://github.com/kaleidawave/enum_variant_type/blob/1908e53bbaf7930bc388328235e285470eb0fb36/src/lib.rs#L645

However your comment here is actually very good, I can think of some use cases where I would want to implement existing marker traits to all variants. And if I wrote the `MyTrait` trait myself and it didn't generate it, it would have the same effect. I think I will modify the name to `implement_marker_traits`, take the arguments as being references and remove the one argument guard (as I agree makes no sense if implementing over existing 😄 )

Reply:
ah! my mistake -- didn't realize the marker trait was also generated

> I think I will modify the name to implement_marker_traits, take the arguments as being references and remove the one argument guard (as I agree makes no sense if implementing over existing smile )

cool :v:; and ya the 1 argument guard makes sense in the "let's generate a marker trait" case :stuck_out_tongue: